hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/data/HoodieJavaRDD.java

```diff
@@ -120,6 +123,26 @@ public int getNumPartitions() {
     return rddData.getNumPartitions();
   }
 
+  @Override
+  public int deduceNumPartitions() {
+    // for source rdd, the partitioner is None
+    final Optional<Partitioner> partitioner = rddData.partitioner();
+    if (partitioner.isPresent()) {
+      int partPartitions = partitioner.get().numPartitions();
+      if (partPartitions > 0) {
+        return partPartitions;
+      }
+    }
+
+    if (SQLConf.get().contains(SQLConf.SHUFFLE_PARTITIONS().key())) {
+      return SQLConf.get().defaultNumShufflePartitions();
+    } else if (rddData.context().conf().contains("spark.default.parallelism")) {
```

Comment:
The Java context may never has this config option right?

Reply:
do you mean javaSparkContext? it contain, I have update the ut conf

Reply:
Why the Java client contains a Spark config option, that is not kind of reasonable.

Reply:
@danny0405 the java rdd  is wrap spark rdd, and this code will running in spark driver, will contain context

Reply:
Oh, my mistake, it's a RDD.